src/gas/fetchGasEstimatesViaEthFeeHistory/BlockFeeHistoryDatasetFetcher.ts

```diff
@@ -22,12 +22,39 @@ export default class BlockFeeHistoryDatasetFetcher {
     return this.fetch({ numberOfBlocks: 20_000 });
   }
 
+  forMediumRange() {
+    return this.fetch({ numberOfBlocks: 200, percentiles: [10, 95] });
+  }
+
   forSmallRange() {
     return this.fetch({ numberOfBlocks: 5, percentiles: [10, 20, 30] });
   }
 
+  forTinyRange() {
+    return this.fetch({
+      numberOfBlocks: 2,
+      percentiles: [50],
+    });
+  }
+
+  forTinyRangeWithPending() {
+    return this.fetch({
+      numberOfBlocks: 2,
+      endBlock: 'pending',
+      percentiles: [50],
+    });
+  }
+
+  forLatest() {
+    return this.fetch({
+      numberOfBlocks: 1,
+      percentiles: [10, 95],
+    });
+  }
+
   private fetch<T extends number = never>(args: {
     numberOfBlocks: number;
+    endBlock?: BN | 'pending';
```

Comment:
`BN | 'pending'` this looks bit weird

Reply:
Sure — this means that the `endBlock` option can either be an instance of the `BN` class or a literal string which _must_ be "pending".